### PR TITLE
feat(Group): Add group users preview

### PR DESCRIPTION
**Pull request by lgmarchi**
### 1. Add group users preview

https://user-images.githubusercontent.com/63157656/235525806-6eaf93f3-34b8-4bb1-a22e-d191d6ec2552.mov

- 

### Which issue(s) this PR fixes 🔨

- Resolve #

### CODE REVIEW

**Review by sdwoodbury — COMMENTED** [review]
State already has all identities for all participants in every conversation and all friends. Given that, is it necessary to obtain the identities again from `warp_runner`?
image


Also, we may want to call the `users` `participants` instead, to match the naming convention from `state::chats::Chat`. Though I know that there are some elements which use the name `users` instead.
image

**Comment by phillsatellite** [comment]
Hi Lucas! Found a small bug that I could not replicate on dev branch 

The File browser seems to be opening after you start a new chat. 

This was replicated on Windows, MacOS, and Linux by just adding a user and creating a chat with them. 

https://user-images.githubusercontent.com/93608357/235708045-c033e395-4351-4bbe-8fd1-12117d32aa3c.mov

**Comment by lgmarchi** [comment]
> State already has all identities for all participants in every conversation and all friends. Given that, is it necessary to obtain the identities again from `warp_runner`? image
> 
> Also, we may want to call the `users` `participants` instead, to match the naming convention from `state::chats::Chat`. Though I know that there are some elements which use the name `users` instead. image

Yes Stu, because it has identities for user friends, if Aser A, is not friend of User C and D in group, will not be possible to get their identities.

**Comment by lgmarchi** [comment]
> Hi Lucas! Found a small bug that I could not replicate on dev branch
> 
> The File browser seems to be opening after you start a new chat.
> 
> This was replicated on Windows, MacOS, and Linux by just adding a user and creating a chat with them.
> 
>  Screen.Recording.2023-05-02.at.11.09.24.AM.mov

I am looking on it my friend.

**Comment by sdwoodbury** [comment]
> > State already has all identities for all participants in every conversation and all friends. Given that, is it necessary to obtain the identities again from `warp_runner`? image
> > Also, we may want to call the `users` `participants` instead, to match the naming convention from `state::chats::Chat`. Though I know that there are some elements which use the name `users` instead. image
> 
> Yes Stu, because it has identities for user friends, if Aser A, is not friend of User C and D in group, will not be possible to get their identities.

not true. see below 
image

image

`init_conversation` converts all the Conversation's `DID`s to identities and those get sent to State. Even for non-friends.

**Comment by sdwoodbury** [comment]
I just checked to make sure that `MessageEvent::RecipientAdded` also updates State::Identities. Looks like all identities are already in state and don't need to be fetched again from Warp.

**Comment by sdwoodbury** [comment]
@phillsatellite  @lgmarchi  the file dialog was related to the panic handler. I removed that from the panic handler and I removed calls to `unwrap()` from `group_user.rs`, which may have been causing the panic.

**Comment by stavares843** [comment]
linter is failing 🔨

File: ui/src/main.rs [hunk 1]
```diff
@@ -162,25 +162,6 @@ fn main() {
...
-                "uplink-crash-report_{}-{}-{}_{}:{}:{}.txt",
-                time.year(),
-                time.month(),
-                time.day(),
-                time.hour(),
-                time.minute(),
-                time.second()
-            );
-            let _ = fs::write(p.join(file_name), crash_report);
-        }
```
Comment (stavares843) [inline ui/src/main.rs:183]:
```
warning: unused import: `rfd::FileDialog`
  --> ui/src/main.rs:30:5
   |
30 | use rfd::FileDialog;
   |     ^^^^^^^^^^^^^^^


```

Comment (stavares843) [inline ui/src/main.rs:183]:
```
warning: unused imports: `Datelike`, `Local`, `Timelike`
 --> ui/src/main.rs:6:14
  |
6 | use chrono::{Datelike, Local, Timelike};
  |              ^^^^^^^^  ^^^^^  ^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

```

**Comment by stavares843** [comment]
linter is failing, regarding not used imports 🔨

**Review by stavares843 — COMMENTED** [review]
Captura de ecrã 2023-05-12, às 23 41 35

we need to improve where click because atm is a bit random we need to click in specific spots for that view to appear, maybe we could add the whole section clickable for that

Captura de ecrã 2023-05-12, às 23 42 11

example

**Comment by sdwoodbury** [comment]
> Captura de ecrã 2023-05-12, às 23 41 35
> 
> we need to improve where click because atm is a bit random we need to click in specific spots for that view to appear, maybe we could add the whole section clickable for that
> Captura de ecrã 2023-05-12, às 23 42 11
> 
> example

After reading this, I don't understand what the problem is or which view isn't appearing.

**Review by sdwoodbury — APPROVED** [review]

**Comment by stavares843** [comment]
what i meant was, when you click in the top tab, it has a very small click area where the click actually makes the group list appear


so i was suggesting that maybe we could apply the click area in the whole section instead

**Pull request merged**